solidity/test/rewards/TestRewardUpgrades.js

```diff
@@ -0,0 +1,222 @@
+const { accounts, contract, web3 } = require("@openzeppelin/test-environment")
+const { createSnapshot, restoreSnapshot } = require("../helpers/snapshot.js")
+const { expectRevert, time } = require("@openzeppelin/test-helpers")
+
+const KeepToken = contract.fromArtifact('KeepToken')
+const RewardsStub = contract.fromArtifact("RewardsStub");
+const NewRewardsStub = contract.fromArtifact("NewRewardsStub");
+
+const BN = web3.utils.BN
+const chai = require('chai')
+chai.use(require('bn-chai')(BN))
+const expect = chai.expect
+
+describe.only("Rewards/Upgrades", () => {
+    const owner = accounts[0]
+    const thirdParty = accounts[1]
+    const beneficiary = accounts[3]
+
+    const intervalWeights = [5, 10, 15, 20]
+    const totalRewards = 1000000
+    const minimumKeepsPerInterval = 1
+    const termLength = 1000
+    let timestamps
+    let firstIntervalStart
+
+    let token, rewards, newRewards
+
+    before(async () => {
+        token = await KeepToken.new({ from: owner })
+        newRewards = await NewRewardsStub.new()
+
+        firstIntervalStart = await time.latest()
+        timestamps = [
+            101, 150,    // 2 keep in interval 0
+            1100,        // 1 keep in interval 1
+            2200, 2201   // 2 keeps in interval 2
+        ].map((t) => firstIntervalStart.addn(t).toNumber())
+
+        rewards = await RewardsStub.new(
+            token.address,
+            minimumKeepsPerInterval,
+            firstIntervalStart,
+            intervalWeights,
+            timestamps,
+            termLength,
+            {from: owner}
+         )
+         await token.approveAndCall(
+            rewards.address,
+            totalRewards,
+            "0x0",
+            {from: owner}
+        )
+        await rewards.markAsFunded({from: owner})
+    })
+
+    beforeEach(async () => {
+        await createSnapshot()
+    })
+
+    afterEach(async () => {
+        await restoreSnapshot()
+    })
+
+    describe("upgrades", async () => {
+        it("can be initiated only by contract owner", async () => {            
+            await expectRevert(
+                rewards.initiateRewardsUpgrade(
+                    newRewards.address,
+                    {from: thirdParty}
+                ),
+                "Ownable: caller is not the owner"
+            )
+        })
+
+        it("can be finalized only by contract owner", async () => {
+            await rewards.initiateRewardsUpgrade(
+                newRewards.address,
+                {from: owner}
+            )
+            await expectRevert(
+                rewards.finalizeRewardsUpgrade({from: thirdParty}),
+                "Ownable: caller is not the owner."
+            )            
+        })
+
+        it("can not be finalized without initiating first", async () => {
```

Comment:
I think it should be written "[cannot](https://www.grammarly.com/blog/cannot-or-can-not/#:~:text=Can't%20is%20a%20contraction%20of%20cannot%2C%20and%20it's%20best,.%20.%20.%20but%20also.%E2%80%9D)" in these tests.